### PR TITLE
Improve Data Playground example

### PR DESCRIPTION
**Pull request by lcaggio**
- Add support to KMS
- Add dedicated SA
- Add firewall rule to SSH to the VM
- Add BQ
- Update diagram

### CODE REVIEW

File: examples/data-solutions/data-playground/README.md [hunk 1]
```diff
@@ -10,34 +10,33 @@ This is the high level diagram:
...
+    - VPC network with a default subnet and CloudNat
     - Firewall rules for [SSH access via IAP](https://cloud.google.com/iap/docs/using-tcp-forwarding) and open communication within the VPC
-- Vertex AI notebook
-  - One Jupyter lab notebook instance with public access
-- GCS
-    - One bucket initial bucket
-
-
+- Vertex AI Workbench notebook with a specialized Service Account
+- One bucket bucket
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/README.md]: Do you mean "one bucket" ?

File: examples/data-solutions/data-playground/outputs.tf [hunk 2]
```diff
@@ -14,12 +14,15 @@
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/outputs.tf:14]: Would it make sense to include the created dataset as well in the output. Might be useful.

File: examples/data-solutions/data-playground/main.tf [hunk 3]
```diff
@@ -71,27 +79,73 @@ module "vpc-firewall" {
   project_id   = module.project.project_id
   network      = module.vpc.name
   admin_ranges = [var.vpc_config.ip_cidr_range]
+  custom_rules = {
+    #TODO Remove and rely on 'ssh' tag once issues/9273 is fixed
+    ("${var.prefix}-iap") = {
+      description          = "Enable SSH from IAP on Notboks."
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/main.tf:85]: Notebooks

File: examples/data-solutions/data-playground/README.md [hunk 4]
```diff
@@ -10,34 +10,33 @@ This is the high level diagram:
...
-| vpc                                                                                                                                                   | VPC Network name        |
-| notebook                                                                                                                                              | Vertex AI notebook name |
+
+| name | description | sensitive |
+|---|---|:---:|
+| [bucket](outputs.tf#L15) | GCS Bucket URL. |  |
+| [notebook](outputs.tf#L20) | Vertex AI notebook details. |  |
+| [project](outputs.tf#L28) | Project id |  |
+| [vpc](outputs.tf#L33) | VPC Network |  |
+
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/README.md:66]: To add the dataset in the output summary, if it gets added in outputs.tf

File: examples/data-solutions/data-playground/main.tf [hunk 5]
```diff
@@ -71,27 +79,73 @@ module "vpc-firewall" {
   project_id   = module.project.project_id
   network      = module.vpc.name
   admin_ranges = [var.vpc_config.ip_cidr_range]
+  custom_rules = {
+    #TODO Remove and rely on 'ssh' tag once issues/9273 is fixed
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/main.tf:83]: Does this comment refer to an issue in the Fabric framework on Github? I don't see one with that number in the list https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/issues

File: examples/data-solutions/data-playground/main.tf [hunk 6]
```diff
@@ -104,10 +158,17 @@ resource "google_notebooks_instance" "playground" {
   install_gpu_driver = true
   boot_disk_type     = "PD_SSD"
   boot_disk_size_gb  = 110
+  disk_encryption    = try(local.service_encryption_keys.compute != null, false) ? "CMEK" : "GMEK"
+  kms_key            = try(local.service_encryption_keys.compute, null)
 
-  no_public_ip    = false
+  no_public_ip    = true
```
Comment (aymanfarhat) [inline examples/data-solutions/data-playground/main.tf:164]: [just double checking] no_public_ip is set to true, does that mean we won't be able to access the notebook's web UI unless we're connected to the same VPC network correct? Might be useful to include that as a quick comment in the README.

**Review by aymanfarhat — CHANGES_REQUESTED** [review]
That's a great addition, thank you @lcaggio it's now a much more useful example. I added a couple of comments and small requests but overall LGTM.

File: examples/data-solutions/data-playground/main.tf (hunk 5 above)
Comment (lcaggio) [reply examples/data-solutions/data-playground/main.tf:83]: The google provider issue: https://github.com/hashicorp/terraform-provider-google/issues/9273

File: examples/data-solutions/data-playground/main.tf (hunk 6 above)
Comment (lcaggio) [reply examples/data-solutions/data-playground/main.tf:164]: We have no_proxy_access = false, you'll be able to connect to the UI

File: examples/data-solutions/data-playground/README.md (hunk 4 above)
Comment (lcaggio) [reply examples/data-solutions/data-playground/README.md:66]: Added

File: examples/data-solutions/data-playground/main.tf (hunk 3 above)
Comment (lcaggio) [reply examples/data-solutions/data-playground/main.tf:85]: Done

File: examples/data-solutions/data-playground/outputs.tf (hunk 2 above)
Comment (lcaggio) [reply examples/data-solutions/data-playground/outputs.tf:14]: Done

**Review by aymanfarhat — APPROVED** [review]

**Pull request merged**